components/scream/src/physics/rrtmgp/tests/generate_baseline.cpp

```diff
@@ -0,0 +1,137 @@
+    int nlev = sw_flux_up_ref.dimension[1];
+    int nlay = nlev - 1;
+
+    // Read in dummy Garand atmosphere; if this were an actual model simulation,
+    // these would be passed as inputs to the driver
+    // NOTE: set ncol to size of col_flx dimension in the input file. This is so
+    // that we can compare to the reference data provided in that file. Note that
+    // this will copy the first column of the input data (the first profile) ncol
+    // times. We will then fill some fraction of these columns with clouds for
+    // the test problem.
+    real2d p_lay ("p_lay", ncol, nlay);
+    real2d t_lay ("t_lay", ncol, nlay);
+    real2d p_lev ("p_lev", ncol, nlay+1);
+    real2d t_lev ("t_lev", ncol, nlay+1);
+    GasConcs gas_concs;
+    read_atmos(inputfile, p_lay, t_lay, p_lev, t_lev, gas_concs, ncol);
+
+    // Initialize the RRTMGP interface; this will read in the k-distribution
+    // data that contains information about absorption coefficients for gases
+    std::cout << "rrtmgp_initialize..." << std::endl;
+    rrtmgp::rrtmgp_initialize(gas_concs);
+
+    // Setup dummy all-sky problem
+    int nswbands = 16;
+    real2d sfc_alb_dir ("sfc_alb_dir", nswbands, ncol);
+    real2d sfc_alb_dif ("sfc_alb_dif", nswbands, ncol);
+    real1d mu0 ("mu0", ncol);
+    real2d lwp ("lwp", ncol, nlay);
+    real2d iwp ("iwp", ncol, nlay);
+    real2d rel ("rel", ncol, nlay);
+    real2d rei ("rei", ncol, nlay);
+    rrtmgpTest::dummy_atmos(
+        inputfile, ncol, p_lay, t_lay,
+        sfc_alb_dir, sfc_alb_dif, mu0,
+        lwp, iwp, rel, rei
+    );
+
+    // Setup flux outputs; In a real model run, the fluxes would be
+    // input/outputs into the driver (persisting between calls), and
+    // we would just have to setup the pointers to them in the
+    // FluxesBroadband object
+    real2d sw_flux_up ("sw_flux_up" ,ncol,nlay+1);
+    real2d sw_flux_dn ("sw_flux_dn" ,ncol,nlay+1);
+    real2d sw_flux_dn_dir("sw_flux_dn_dir",ncol,nlay+1);
+    real2d lw_flux_up ("lw_flux_up" ,ncol,nlay+1);
+    real2d lw_flux_dn ("lw_flux_dn" ,ncol,nlay+1);
+
+    // Run RRTMGP standalone codes and compare with AD run
+    // Do something interesting here...
+    // NOTE: these will get replaced with AD stuff that handles these
+    std::cout << "rrtmgp_main..." << std::endl;
+    rrtmgp::rrtmgp_main(
+        p_lay, t_lay, p_lev, t_lev, gas_concs,
+        sfc_alb_dir, sfc_alb_dif, mu0,
+        lwp, iwp, rel, rei,
+        sw_flux_up, sw_flux_dn, sw_flux_dn_dir,
+        lw_flux_up, lw_flux_dn
+    );
+
+    // Write fluxes
+    std::cout << "write_fluxes..." << std::endl;
+    rrtmgpTest::write_fluxes(
+        baseline, 
+        sw_flux_up, sw_flux_dn, sw_flux_dn_dir,
+        lw_flux_up, lw_flux_dn
+    );
+
+    // Clean up from test; this is probably not necessary, these things
+    // should be deallocated when they fall out of scope, but we should be
+    // good citizens and clean up our mess.
+    p_lay.deallocate();
```

Comment:
It might be nice to know, as a matter of practice, whether these deallocations are actually necessary. C++ libraries are so full of leaky abstractions that it's made us all pretty superstitious about this stuff.

Reply:
Ah yes, I’ll do some testing to confirm. I *did* have some problems with YAKL throwing segfaults on exit though if I didn’t clean things up appropriately at one point. This may not be a case of that, but, I suspect it is.